### PR TITLE
Remove anything related to ReferenceOutputAssembly from being include…

### PR DESCRIPTION
**Pull request by chandramouleswaran**
This PR addresses - https://github.com/NuGet/Home/issues/1540

Basically, we do not want to include assemblies of project references which are only used to help resolving build dependencies.

For example: I might have a .proj file which does say VHD creation but I do not want its output to be a part of the package.

This can be done using ReferenceOutputAssembly being false which is a standard now. I have also added tests for the same.

### CODE REVIEW

File: src/NuGet.Clients/NuGet.CommandLine/Commands/ProjectFactory.cs [hunk 1]
```diff
@@ -13,6 +13,11 @@
 
 namespace NuGet.CommandLine
 {
+    using Microsoft.Build.Evaluation;
+    using Microsoft.Build.Execution;
+
+    using Console = System.Console;
+
```
Comment (yishaigalatzer) [inline src/NuGet.Clients/NuGet.CommandLine/Commands/ProjectFactory.cs:20]: move the using statements outside of the namespace

**Comment by yishaigalatzer** [comment]
Nice! Thanks for taking the time to include tests.

@deepakaravindr assigned the bug and the PR for you to review and verify before merging.

**Comment by deepakaravindr** [comment]
Will be closed when the PR https://github.com/NuGet/NuGet.Client/pull/68 is closed

**Comment by deepakaravindr** [comment]
Fixed with https://github.com/NuGet/NuGet.Client/commit/f0cde66f22012083463b853fc3563165a4fea177 and https://github.com/NuGet/NuGet.Client/commit/f0cde66f22012083463b853fc3563165a4fea177

**Pull request closed without merging**